### PR TITLE
UML-117-API-LAYER Added ecs service

### PR DESCRIPTION
**Pull request by SeemaMenon**

### CODE REVIEW

File: terraform_environment/shared_data_sources.tf [hunk 1]
```diff
@@ -56,3 +56,13 @@ data "aws_ecr_repository" "use_an_lpa_front_app" {
...
   name     = "use_an_lpa/front_app"
 }
+data "aws_ecr_repository" "use_an_lpa_api_app" {
+  provider = "aws.management"
+  name     = "use_an_lpa/api_app"
+}
+
+data "aws_ecr_repository" "use_an_lpa_api_web" {
+  provider = "aws.management"
+  name     = "use_an_lpa/api_app"
```
Comment (nsmithuk) [inline terraform_environment/shared_data_sources.tf]: I think this should be `api_web`

Comment (SeemaMenon) [reply terraform_environment/shared_data_sources.tf]: made the change

File: terraform_environment/api_ecs.tf [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    actions = [
+      "dynamodb:*",
+    ]
+
+    resources = ["${aws_dynamodb_table.viewer_codes_table.arn}"]
+  }
+}
+
+//-----------------------------------------------
+// Viewer ECS Service Task Container level config
```
Comment (nsmithuk) [inline terraform_environment/api_ecs.tf]:
```suggestion
// API ECS Service Task Container level config
```

File: terraform_environment/api_ecs.tf [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+
+    routing_policy = "MULTIVALUE"
+  }
+
+  health_check_custom_config {
+    failure_threshold = 1
+  }
+}
+
+resource "aws_service_discovery_private_dns_namespace" "internal" {
```
[the remark this answers is not in the document]
Comment (SeemaMenon) [reply terraform_environment/api_ecs.tf]: Done the changes

**Pull request merged**